compiler-rt/lib/sanitizer_common/sanitizer_fuchsia.cpp

```diff
@@ -357,7 +406,7 @@ void *MmapAlignedOrDieOnFatalError(uptr size, uptr alignment,
 }
 
 void UnmapOrDie(void *addr, uptr size) {
-  UnmapOrDieVmar(addr, size, _zx_vmar_root_self());
+  UnmapOrDieVmar(addr, size, gSanitizerHeapVmar);
```

Comment:
This needs to use the fallback logic in case the root vmar was used, as above.

Reply:
I see, thanks. The way I saw the diff that was hard to find, probably just my lack of familiarity with the GH code review UI.